app/screens/channel_info/channel_info.js

```diff
@@ -234,58 +228,92 @@ export default class ChannelInfo extends PureComponent {
         );
     });
 
-    handleDeleteOrLeave = preventDoubleTap((eventType) => {
+    handleLeave = preventDoubleTap(() => {
+        const title = {id: t('mobile.channel_info.alertTitleLeaveChannel'), defaultMessage: 'Leave {term}'};
+        const message = {
+            id: t('mobile.channel_info.alertMessageLeaveChannel'),
+            defaultMessage: 'Are you sure you want to leave the {term} {name}?',
+        };
+        const onPressAction = () => {
+            this.props.actions.leaveChannel(this.props.currentChannel, true).then(() => {
+                this.close();
+            });
+        };
+        this.alertAndHandleYesAction(title, message, onPressAction);
+    });
+
+    handleDelete = preventDoubleTap(() => {
+        const channel = this.props.currentChannel;
+        const title = {id: t('mobile.channel_info.alertTitleDeleteChannel'), defaultMessage: 'Archive {term}'};
+        const message = {
+            id: t('mobile.channel_info.alertMessageDeleteChannel'),
+            defaultMessage: 'Are you sure you want to archive the {term} {name}?',
+        };
+        const onPressAction = async () => {
+            const result = await this.props.actions.deleteChannel(channel.id);
+            if (result.error) {
+                alertErrorWithFallback(
+                    this.context.intl,
+                    result.error,
+                    {
+                        id: t('mobile.channel_info.delete_failed'),
+                        defaultMessage: "We couldn't archive the channel {displayName}. Please check your connection and try again.",
```

Comment:
are we sure this is connection related? and if so why then we would try and call `getChannel`?

Reply:
you make a good point. Should we change this message to something more generic?